pyaedt/modules/Material.py

```diff
@@ -272,579 +492,507 @@ def desktop(self):
         """ """
         return self._parent._desktop
 
-    def __init__(self, parent):
+    def __init__(self, parent, name, props=None):
         self._parent = parent
-        self.name = ""
-        self.origin = ""
-        self.property = defaultdict(MatProperty)
-        self.thermal_material_type="Solid"
-        for el in MatProperties.aedtname:
-            self.property[el].property_value[0].value = MatProperties.get_defaultvalue(aedtname=el)
-            self.property[el].unit = MatProperties.get_defaultunit(aedtname=el)
-            self.property[el].type = "simple"
+        self.name = name
+        self.coordinate_system = ""
 
+        if props:
+            self._props = props
+        else:
+            self._props = OrderedDict()
+        if "CoordinateSystemType" in self._props:
+            self.coordinate_system = self._props["CoordinateSystemType"]
+        else:
+            self._props["CoordinateSystemType"] = "Cartesian"
+            self.coordinate_system = "Cartesian"
+        if "BulkOrSurfaceType" in self._props:
+            self.bulkorsurface = self._props["BulkOrSurfaceType"]
+        else:
+            self._props["BulkOrSurfaceType"] = 1
+        if "ModTime" in self._props:
+            self._modtime = self._props["ModTime"]
+            del self._props["ModTime"]
+        if "LibLocation" in self._props:
+            self.lib_location = self._props["LibLocation"]
+            del self._props["LibLocation"]
+        if "ModSinceLib" in self._props:
+            self.mod_since_lib = self._props["ModSinceLib"]
+            del self._props["ModSinceLib"]
 
-    def get_number(self, fullname):
-        """
 
-        Parameters
-        ----------
-        fullname :
-            
+    @aedt_exception_handler
+    def _get_args(self, props=None):
+        if not props:
+            props = self._props
+        arg = ["NAME:" + self.name]
+        dict2arg(props, arg)
+        return arg
+
+    def _update_props(self, propname, provpavlue, update_aedt=True):
```

Comment:
Protected methods don't need well written docstrings, but we need at least a one liner.